src/logic/graphMapping.ts

```diff
@@ -0,0 +1,115 @@
+
+import { Graph, MappedEdge, MappedNode } from './commonInterfaces';
+
+const LABEL_ALPHABET = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+
+export class Edge {
+  label?: number;
+  from: number;
+  to: number;
+  equals = (other: Edge) => {
+    return this.label === other.label && (
+      (this.from === other.from && this.to === other.to) ||
+      (this.from === other.to && this.to === other.from)
+    )
+  };
+  constructor(from: number, to: number, label?: number) {
+    this.from = from;
+    this.to = to;
+    this.label = label;
+  }
+};
+
+/**
+ * Finds all edges in graph matrix. This maybe inefficient (complexity of
+ * O(n^3)), but for this purpose it's ok. Tested up to 2016 edges. For
+ * that amount takes < 1s. Maybe a bottleneck in the future.
```

Comment:
... **Possible** bottleneck in the future.

Reply:
I see. So you are reading comments :D